common/usermode.c

```diff
@@ -237,6 +234,6 @@ void *__user_text mmap(void *va, unsigned long order) {
     return _ptr(sys_mmap(va, order));
 }
 
-void __user_text munmap(void *va, unsigned long order) {
-    sys_munmap(va, order);
+int __user_text munmap(void *va) {
+    return sys_munmap(va);
```

Comment:
This change is unfortunate, as this will only unmap the very first page. :(

What about passing a `length` argument to specify how much to unmap? This would allow to partially unmap regions as fits, much like munmap() does on UNIX systems?

Reply:
Uh, yes that's true. I will change this as recommended.

Reply:
As noted throughout the series, this should be a follow-up change.